src/xtgeo/xyz/_xyz.py

```diff
@@ -79,39 +97,68 @@ class XYZ:
 
     Additional columns are possible but certainly not required. These are free
     attributes with user-defined names. These names (with data-type) are
-    stored in ordered dict: self._attrs as {"somename": "type", ...}
+    stored in ordered dict: self._attrs as {"somename": "type", ...}.
     """
 
     @allow_deprecated_init
     def __init__(
         self,
-        spesification: Optional[Any] = None,
+        values: Optional[Union[list, np.ndarray, pd.DataFrame]] = None,
         xname: Optional[str] = "X_UTME",
-        yname: Optional[str] = "X_UTME",
-        zname: Optional[str] = "X_UTME",
+        yname: Optional[str] = "Y_UTMN",
+        zname: Optional[str] = "Z_TVDSS",
         pname: Optional[str] = "POLY_ID",
-        dfr: Optional[pd.DataFrame] = None,
+        name: Optional[str] = "unknown",
```

Comment:
It would be easier to evaluate this Refactoring of xyz in a separate commit. Also it seems to be a bit of a work in progress. In order to get things moving, It would be good to have one PR for the addition of `allow_deprecated_init` and the `__init__` changes related to #524 which we could review and add right now, and then a separate work in progress PR for the remaining changes.